x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/TDigestParser.java

```diff
@@ -18,18 +18,78 @@
 import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
 import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.CENTROIDS_NAME;
 import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.COUNTS_NAME;
+import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.MAX_FIELD_NAME;
+import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.MIN_FIELD_NAME;
+import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.SUM_FIELD_NAME;
+import static org.elasticsearch.xpack.analytics.mapper.TDigestFieldMapper.TOTAL_COUNT_FIELD_NAME;
 
 public class TDigestParser {
 
     private static final ParseField COUNTS_FIELD = new ParseField(COUNTS_NAME);
     private static final ParseField CENTROIDS_FIELD = new ParseField(CENTROIDS_NAME);
+    private static final ParseField TOTAL_COUNT_FIELD = new ParseField(TOTAL_COUNT_FIELD_NAME);
+    private static final ParseField SUM_FIELD = new ParseField(SUM_FIELD_NAME);
+    private static final ParseField MAX_FIELD = new ParseField(MAX_FIELD_NAME);
+    private static final ParseField MIN_FIELD = new ParseField(MIN_FIELD_NAME);
 
     /**
      * A parsed histogram field, can represent either a T-Digest
      * @param centroids the centroids, guaranteed to be distinct and in increasing order
      * @param counts the counts, guaranteed to be non-negative and of the same length as the centroids array
      */
-    public record ParsedHistogram(List<Double> centroids, List<Long> counts) {}
+    public record ParsedHistogram(List<Double> centroids, List<Long> counts, Long count, Double sum, Double min, Double max) {
+        @Override
+        public Double max() {
+            if (max != null) {
+                return max;
+            }
+            if (centroids != null && centroids.isEmpty() == false) {
+                return centroids.get(centroids.size() - 1);
+            }
+            return Double.NaN;
```

Comment:
Is there a reason to always calculate this lazily?  As if not I think this would be easier to reason about if the record used primitive types (eg `long` rather than `Long`) and we had a factory method that calculated summary values if they are not present.